### DISCUSSION TITLE
[@types/apple-mapkit-js-browser]: replace type 'SearchOptions' with 'SearchAutocompleteOptions;' on Search.autocomplete's  'option' formal parameter

### DISCUSSION THREAD

**Discussion by yllaw** [DefinitelyTyped/DefinitelyTyped#59330] [Issues with a @types package]
Class `Search` function's `autocomplete` formal parameter `options` is using type `SearchOptions` instead of `SearchAutoCompleteOptions`

Docs for reference:
https://developer.apple.com/documentation/mapkitjs/mapkit/search/2974016-autocomplete

[a comment by an automated account is not shown]

**Comment by moritzsternemann** [comment]
@yllaw The interface definition of `SearchAutocompleteOptions` contains the `limitToCountries` property (see [types/apple-mapkit-js-browser/index.d.ts#L1935](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/5a6e1f80e21350d39c9f04af96763309741cd3c6/types/apple-mapkit-js-browser/index.d.ts#L1935), added ~8 Months ago). Maybe you are using an older version 🤔

**Reply by yllaw** [reply]
Ah the problem is actually the Class `Search` function `autocomplete` formal parameter `options`  is using type `SearchOptions` instead of `SearchAutoCompleteOptions`

see [types/apple-mapkit-js-browser/index.d.ts#L1846](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/5a6e1f80e21350d39c9f04af96763309741cd3c6/types/apple-mapkit-js-browser/index.d.ts#L1846)
[Apple docs for Search.autocomplete](https://developer.apple.com/documentation/mapkitjs/mapkit/search/2974016-autocomplete)

I have edited the original post to reflect the problem change

**Reply by moritzsternemann** [reply]
Good catch! Maybe you can prepare a PR to fix that, otherwise I can do it later today 🙂
